### DISCUSSION TITLE
Extra arrows not matching Dexcom arrows

### DISCUSSION THREAD

**Discussion by CollinCMac** [Navid200/xDrip#154] [Dexcom]
Hello. I'm very non-technical FYI. I just managed to get the xdrip+ working on my Android phone and Fitbit Versa 2 watch. We have Dexcom G6. I'm so very thankful that it works but the arrows on the Xdrip+ and my watch do not match the arrows on the Dexcom information. See pix.  Normally I could just ignore it but a few times overnight it showed double arrows down when on Dexcom it was level. This caused falling fast alerts when that was not the case. There were also other times where it showed it going straight up when it was level. Just overall it's not showing the arrows going in sync. FYI I did shut off the predicted low alert and it has not changed anything. It's not the low predictions or the low alarms that are the problem It's the overall problem of the arrows going in different directions than the decks come data. I hope that makes sense.  Anyone have any suggestion on how to get the arrows to sync up? The numbers are correct just not the arrows.  Thank you 
[PXL_20211212_035354459](https://user-images.githubusercontent.com/56234304/145720894-b43ba29f-b1e0-4f0f-a0f2-163d45469d4c.jpg)
![Screenshot_20211211-225348](https://user-images.githubusercontent.com/56234304/145720897-fbbcf753-0046-42d5-bf71-e45e2267ec00.png)
![Screenshot_20211211-225333](https://user-images.githubusercontent.com/56234304/145720898-aa7cd428-f367-4edc-b385-3a3798e46da1.png)

**Comment by Navid200** [comment]
As a first step, please enable graphical trend arrow.
To do that, in xDrip, hold your finger on the xDrip icon (blood drop), for a few seconds.
A form will pop up.  Enable "Show graphical trend arrow".

After doing that, please let us know here if you still experience the same problem.

**Comment by CollinCMac** [comment]
Hello.  Attached are the photos immediately after following the directions below when the Dexcom updated ...
The arrows are still out of sync.
Thank you
Lisa 






 
 
  On Sun, Dec 12, 2021 at 3:28 PM, ***@***.***> wrote:   


As a first step, please enable graphical trend arrow.
To do that, in xDrip, hold your finger on the xDrip icon (blood drop), for a few seconds.
A form will pop up. Enable "Show graphical trend arrow".

After doing that, please let us know here if you still experience the same problem.

**Reply by Navid200** [reply]
I see no attachments.

**Comment by CollinCMac** [comment]
![Screenshot_20211212-153824](https://user-images.githubusercontent.com/56234304/145729235-9f688330-fe66-462f-9aa0-eb644e97edbd.png)
![Screenshot_20211212-153817](https://user-images.githubusercontent.com/56234304/145729236-d0a4a2c0-1221-4a8e-b2c1-fc5df8c1e8c2.png)
![Screenshot_20211212-153836](https://user-images.githubusercontent.com/56234304/145729237-942abb57-feb0-4cb0-8fca-cf6cb0b283cf.png)

**Reply by Navid200** [reply]
But, you can see that the current reading is greater than the previous.  So, a flat arrow wouldn't be correct.

**Comment by CollinCMac** [comment]
Okay that is true but it's different than the prediction from Dexcom and they're not matching. For example overnight it showed double arrows down let me show you that one and it wasn't correct. It set off alarms that it was dropping fast and that wasn't correct. The numbers are syncing but the arrows are not.....

I thought I took a picture of that but I did not. Overnight it showed that the number was double arrows down when Dexcom actually showed it level.  It's been this way since I had this app working.

**Comment by CollinCMac** [comment]
I guess I'm just assuming that what Dexcom shows with numbers and arrows xdrip is supposed to just mirror it. Is that not the case? Does xdrip essentially decide its own arrows and which direction it thinks it's going?

**Reply by Navid200** [reply]
There is an ongoing discussion about that:
https://github.com/Navid200/xDrip/discussions/142

There are claims that the Dexcom receiver and the Dexcom app rewrite the history in order to make the graph look smooth.

**Reply by CollinCMac** [reply]
Ok I see.  Ill have to look into that discussion.  Well, thank you for trying to help. I sure do appreciate it

**Reply by jonprogers** [reply]
I believe the difference is that the official dexcom algorithm bases the direction arrow on the smoothed change in BG over about the last 20 minutes whereas the xDrip algorithm bases the direction arrow on the difference between the last two readings.

You can see this clearly in your 3:38 set of pictures, the graph is gently dropping and then rises at the end so for dexcom, over about the last 20 mins, the trend is on average flat but for xDrip the last change was up so the trend is shown as gentle up.

These are both different takes on the same data and are both 'correct' based on their own definition of how they calculate the trend. The dexcom algorithm is good at smoothing out noise and limiting false alarms at the expense of delaying some alarms whereas the xDrip algorithm gives the quickest possible response as there is no low pass filtering but therefore fails to filter out noise in some circumstances.

Personally I tend to look at the xDrip arrow first, then the graph to understand the context around the trend, then maybe also look at how dexcom is interpreting things. It is all great information that makes my guesses around treatment more and more educated - but unfortunately rarely perfect!!

If only life as a T1D were simple...

**Comment by CollinCMac** [comment]
That makes perfect sense now. Thank you! It really does help to know what's going on with both sets of data. I appreciate you taking the time to respond and explain things so well. Thanks! 😁👍
